server/graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs

```diff
@@ -5,21 +5,39 @@ use graphql_core::standard_graphql_error::{validate_auth, StandardGraphqlError};
 use graphql_core::ContextExt;
 use graphql_types::types::InvoiceLineNode;
 
+use ::serde::Serialize;
 use repository::InvoiceLine;
 use service::auth::{Resource, ResourceAccessRequest};
-use service::invoice_line::outbound_shipment_line::{
-    InsertOutboundShipmentLine as ServiceInput, InsertOutboundShipmentLineError as ServiceError,
+use service::invoice_line::stock_out_line::{
+    InsertOutInvoiceLine as ServiceInput, InsertOutInvoiceLineError as ServiceError, InsertOutType,
 };
 
 use super::{
     LocationIsOnHold, LocationNotFound, NotEnoughStockForReduction,
     StockLineAlreadyExistsInInvoice, StockLineIsOnHold,
 };
 
+#[derive(Enum, Copy, Clone, PartialEq, Eq, Debug, Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum InsertOutTypeNode {
+    OutboundShipment,
+    Prescription,
+}
+
+impl InsertOutTypeNode {
+    pub fn to_domain(self) -> InsertOutType {
+        match self {
+            InsertOutTypeNode::OutboundShipment => InsertOutType::OutboundShipment,
+            InsertOutTypeNode::Prescription => InsertOutType::Prescription,
+        }
+    }
+}
+
 #[derive(InputObject)]
 #[graphql(name = "InsertOutboundShipmentLineInput")]
 pub struct InsertInput {
     pub id: String,
+    pub r#type: Option<InsertOutTypeNode>,
```

Comment:
Sorry Roxy, I wasn't specific enough.

The type in service is great:

https://github.com/openmsupply/open-msupply/blob/98035afb9bc7f335fdddf21415013541f0ad6b2d/server/service/src/invoice_line/stock_out_line/insert/mod.rs#L24-L27

But we don't need one in Graphql, by "pass the type to service in graphql" I mean when calling service, depending on the context, use the type:

```diff
diff --git a/server/graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs b/server/graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs
index 477176520..d29cf308d 100644
--- a/server/graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs
+++ b/server/graphql/invoice_line/src/mutations/outbound_shipment_line/line/insert.rs
@@ -106,7 +106,6 @@ impl InsertInput {
     pub fn to_domain(self) -> ServiceInput {
         let InsertInput {
             id,
-            r#type,
             invoice_id,
             item_id,
             stock_line_id,
@@ -117,7 +116,8 @@ impl InsertInput {
 
         ServiceInput {
             id,
-            r#type: r#type.map(|r#type| r#type.to_domain()),
+            // Or as another parameter to insert_stock_out_line
+            r#type: InsertOutType::OutboundShipment,
             invoice_id,
             item_id,
             stock_line_id,


```